frontend/src/components/App/Home/index.tsx

```diff
@@ -0,0 +1,213 @@
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { useDispatch } from 'react-redux';
+import { useHistory } from 'react-router-dom';
+import helpers from '../../../helpers';
+import { useClustersConf, useClustersVersion } from '../../../lib/k8s';
+import { ApiError, deleteCluster } from '../../../lib/k8s/apiProxy';
+import { Cluster } from '../../../lib/k8s/cluster';
+import { createRouteURL } from '../../../lib/router';
+import { useFilterFunc, useId } from '../../../lib/util';
+import { setConfig } from '../../../redux/actions/actions';
+import { Link, PageGrid, SectionBox, SectionFilterHeader, SimpleTable } from '../../common';
+import RecentClusters from './RecentClusters';
+
+function ContextMenu({ cluster }: { cluster: Cluster }) {
+  const { t } = useTranslation(['settings', 'frequent']);
+  const history = useHistory();
+  const dispatch = useDispatch();
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+  const menuId = useId('context-menu');
+
+  function removeCluster(cluster: Cluster) {
+    deleteCluster(cluster.name || '')
+      .then(config => {
+        dispatch(setConfig(config));
+      })
+      .catch((err: Error) => {
+        if (err.message === 'Not Found') {
+          // TODO: create notification with error message
+        }
+      })
+      .finally(() => {
+        history.push('/');
+      });
+  }
+
+  function handleMenuClose() {
+    setAnchorEl(null);
+  }
+
+  return (
+    <>
+      <IconButton
+        size="small"
+        onClick={event => {
+          setAnchorEl(event.currentTarget);
+        }}
+        aria-haspopup="menu"
+        aria-controls={menuId}
+        aria-label={t('frequent|Actions')}
+      >
+        <Icon icon="mdi:more-vert" />
+      </IconButton>
+      <Menu
+        id={menuId}
+        anchorEl={anchorEl}
+        keepMounted
+        open={Boolean(anchorEl)}
+        onClose={() => {
+          handleMenuClose();
+        }}
+      >
+        <MenuItem
+          onClick={() => {
+            history.push(createRouteURL('cluster', { cluster: cluster.name }));
+            handleMenuClose();
+          }}
+        >
+          <ListItemText>{t('settings|View')}</ListItemText>
+        </MenuItem>
+        <MenuItem
+          onClick={() => {
+            history.push(createRouteURL('settingsCluster', { cluster: cluster.name }));
+            handleMenuClose();
+          }}
+        >
+          <ListItemText>{t('settings|Settings')}</ListItemText>
+        </MenuItem>
+        {helpers.isElectron() && cluster.meta_data?.source === 'dynamic_cluster' && (
+          <MenuItem
+            onClick={() => {
+              removeCluster(cluster);
+              handleMenuClose();
+            }}
+          >
+            <ListItemText>{t('frequent|Delete')}</ListItemText>
+          </MenuItem>
+        )}
+      </Menu>
+    </>
+  );
+}
+
+function ClusterStatus({ error }: { error?: ApiError | null }) {
```

Comment:
This could be in Home/ClusterStatus.tsx

There could be stories for this.

Reply:
I'll try to add a story for the whole Home.